src/editor/codeeditor.h

```diff
@@ -23,6 +23,9 @@ const int TOP_UNUSED_PIXELS_HEIGHT = 4;
 #include<QVector>
 #include<QFont>
 #include<QStatusBar>
+#include<QStringList>
+#include<QCompleter>
+#include"autocodecompleter.h"
 
 
 class QPaintEvent;
```

Comment:
use QT_BEGIN_NAMESPACE and QT_END_NAMESPACE to enclose class declaration

Reply:
It is Petro's code